R/Spectra.R

```diff
@@ -1508,6 +1554,48 @@ setMethod("filterMsLevel", "Spectra", function(object, msLevel. = integer()) {
     object
 })
 
+#' @rdname Spectra
+#'
+#' @export
+setMethod("filterMzRange", "Spectra",
+          function(object, mz = numeric(), msLevel. = unique(msLevel(object))) {
+              if (!.check_ms_level(object, msLevel.))
+                  return(object)
+              mz <- range(mz)
+              object <- addProcessing(object, .peaks_filter_mz_range,
+                                      mz = mz, msLevel = msLevel.)
+              object@processing <- .logging(
+                  object@processing, "Filter: select peaks with an m/z within ",
+                  "[", paste0(mz, collapse = ", "), "]")
+              object
+          })
+
+#' @rdname Spectra
+#'
+#' @export
+setMethod("filterMzValues", "Spectra",
+          function(object, mz = numeric(), tolerance = 0, ppm = 20,
+                   msLevel. = unique(msLevel(object))) {
+              if (!.check_ms_level(object, msLevel.))
+                  return(object)
+              if (is.unsorted(mz)) {
+                  idx <- order(mz)
+                  mz <- mz[idx]
+                  l <- length(mz)
+                  if (length(tolerance) == l)
```

Comment:
Actually, I thought that this would be catched downstream by the `closest` call - but maybe you're right and it might be safer (and also immediate) to test it here.

Reply:
Done in https://github.com/rformassspectrometry/Spectra/commit/927c6ad32f4ce3fa2330713c1774c4874827541c

Reply:
Oh, yes you are right. Duplicating this check seems to be useless. So may we should use the `.check` argument in `closest` to avoid that check as well? (because here it is run once and in `closest` `length(object)` times)